microsoft/testsuites/core/vm_hot_resize.py

```diff
@@ -0,0 +1,44 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from assertpy import assert_that
+
+from lisa import Node, TestCaseMetadata, TestSuite, TestSuiteMetadata
+from lisa.features.resize import Resize
+from lisa.schema import NodeSpace
+from lisa.tools import Lscpu
+
+
+@TestSuiteMetadata(
+    area="vm_hot_resize",
+    category="functional",
+    description="""
+    This test suite tests vm behavior upon resizing without shutting down
+    """,
+)
+class VmHotResize(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case resizes the node and checks if it has the expected capabilities
+        (memory size and core count) after the resize
+
+        Steps:
+        1. Resize vm
+        2. Check the node's core count and memory size against their expected values
+        """,
+        priority=1,
+    )
+    def verify_vm_hot_resize(self, node: Node) -> None:
+        resize = node.features[Resize]
+        expected_vm_capability = resize.resize()
+        self._verify_core_count(node, expected_vm_capability)
```

Comment:
How about also check the cpu count is increased after resize?

Reply:
expected_vm_capability (which is a Nodespace) has no variable or method to return the cpu count. How can I get the expected cpu count?

Reply:
the `core_count` is the cpu count.

Reply:
I thought that the CPU count and core count can differ sometimes

Reply:
Yes, sometime it's confusing and different sometimes. The `core` is the vcpu or hyperthreading here, and the cpu count returned by lscpu should be this value too.

Reply:
done